File: Inventory Management System/src/main/java/com/example/inventorymanagementsystem/DTO/ProductAdminResponse.java
package com.example.inventorymanagementsystem.DTO;

import java.math.BigDecimal;
import java.time.Instant;

public record ProductAdminResponse(
        Long id,
        String sku,
        String name,
        Long categoryId,
        String categoryName,
        Integer quantityOnHand,
        Integer reorderLevel,
        BigDecimal unitCost,
        BigDecimal unitPrice,
        String status,
        String description,
        Integer version,
        Instant createdAt,
        Instant updatedAt
) {
}

File: FrontEnd/frontend-app/src/app/Model/product-admin-response.ts
export interface ProductAdminResponse {
  id: number;
  sku: string;
  name: string;
  categoryId: number;
  categoryName: string;
  quantityOnHand: number;
  reorderLevel: number;
  unitCost: number;
  unitPrice: number;
  status: 'ACTIVE' | 'INACTIVE';
  description: string;
  version: number;
  createdAt: string;
  updatedAt: string;
}